🐙 source-typeform: run up-to-date pipeline [2025-05-31]

airbyteio opened the pull request:
# Update source-typeform

This PR was autogenerated by running `airbyte-ci connectors --name=source-typeform up_to_date --pull`

We've set the `auto-merge` label on it, so it will be automatically merged if the CI pipelines pass.
If you don't want to merge it automatically, please remove the `auto-merge` label.
Please reach out to the Airbyte Connector Tooling team if you have any questions or concerns.


## Operations

- Upgrade the base image to the latest version in metadata.yaml: Successful

- PATCH bump source-typeform version to 1.4.2: Successful

- Build source-typeform docker image for platform(s) linux/amd64, linux/arm64: Successful

- Get dependency updates: Successful

- Create or update pull request on Airbyte repository: Successful

- Add changelog entry: Successful




## Dependency updates

We use [`syft`](https://github.com/anchore/syft) to generate a SBOM for the latest connector version and the one from the PR.
It allows us to spot the dependencies that have been updated at all levels and for all types of dependencies (system, python, java etc.).
Here are the dependencies that have been updated compared to `airbyte/source-typeform:latest`.
Keep in mind that `:latest` does not always match the connector code on the main branch. 
It is the latest released connector image when the head commit of this branch was created.

| Type | Name | State | Previous Version | New Version |
|------|------|-------|-------------|------------------|
| python | CacheControl | updated | 0.13.1 | **0.14.2** |
| python | Jinja2 | updated | 3.1.4 | **3.1.6** |
| python | RapidFuzz | updated | 3.10.1 | **3.11.0** |
| python | airbyte-cdk | updated | 6.10.0 | **6.53.0** |
| python | airbyte_protocol_models_dataclasses | updated | 0.14.1 | **0.15.0** |
| python | anyio | updated | 4.7.0 | **4.9.0** |
| python | attrs | updated | 24.2.0 | **24.3.0** |
| deb | base-files | updated | 12.4+deb12u8 | **12.4+deb12u9** |
| deb | bsdutils | updated | 1:2.38.1-5+deb12u2 | **1:2.38.1-5+deb12u3** |
| python | build | updated | 0.10.0 | **1.2.2.post1** |
| python | cachetools | updated | 5.5.0 | **5.5.1** |
| python | certifi | updated | 2024.8.30 | **2024.12.14** |
| python | charset-normalizer | updated | 3.4.0 | **3.4.1** |
| python | click | updated | 8.1.7 | **8.1.8** |
| python | cryptography | updated | 43.0.3 | **44.0.2** |
| python | filelock | updated | 3.16.1 | **3.17.0** |
| python | h11 | updated | 0.14.0 | **0.16.0** |
| python | httpcore | updated | 1.0.7 | **1.0.9** |
| python | joblib | updated | 1.4.2 | **1.5.1** |
| deb | libblkid1 | updated | 2.38.1-5+deb12u2 | **2.38.1-5+deb12u3** |
| deb | libglib2.0-0 | updated | 2.74.6-2+deb12u4 | **2.74.6-2+deb12u5** |
| deb | libglib2.0-data | updated | 2.74.6-2+deb12u4 | **2.74.6-2+deb12u5** |
| deb | libmount1 | updated | 2.38.1-5+deb12u2 | **2.38.1-5+deb12u3** |
| deb | libopenjp2-7 | updated | 2.5.0-2 | **2.5.0-2+deb12u1** |
| deb | libsmartcols1 | updated | 2.38.1-5+deb12u2 | **2.38.1-5+deb12u3** |
| deb | libsystemd0 | updated | 252.31-1~deb12u1 | **252.33-1~deb12u1** |
| deb | libtiff6 | updated | 4.5.0-6+deb12u1 | **4.5.0-6+deb12u2** |
| deb | libudev1 | updated | 252.31-1~deb12u1 | **252.33-1~deb12u1** |
| deb | libuuid1 | updated | 2.38.1-5+deb12u2 | **2.38.1-5+deb12u3** |
| python | more-itertools | updated | 10.5.0 | **10.6.0** |
| deb | mount | updated | 2.38.1-5+deb12u2 | **2.38.1-5+deb12u3** |
| python | orjson | updated | 3.10.12 | **3.10.15** |
| python | pkginfo | updated | 1.11.2 | **1.12.0** |
| python | platformdirs | updated | 3.11.0 | **4.3.6** |
| python | poetry | updated | 1.6.1 | **1.8.4** |
| python | poetry-core | updated | 1.7.0 | **1.9.1** |
| python | poetry-plugin-export | updated | 1.6.0 | **1.8.0** |
| python | pydantic | updated | 2.10.3 | **2.10.5** |
| python | pydantic_core | updated | 2.27.1 | **2.27.2** |
| binary | python | updated | 3.10.14 | **3.11.11** |
| python | pytz | updated | 2024.1 | **2024.2** |
| python | serpyco-rs | updated | 1.11.0 | **1.13.0** |
| python | trove-classifiers | updated | 2024.10.21.16 | **2025.1.15.22** |
| python | typing_extensions | updated | 4.12.2 | **4.13.2** |
| deb | tzdata | updated | 2024a-0+deb12u1 | **2024b-0+deb12u1** |
| python | tzdata | updated | 2024.2 | **2025.1** |
| python | urllib3 | updated | 2.2.3 | **2.3.0** |
| deb | util-linux | updated | 2.38.1-5+deb12u2 | **2.38.1-5+deb12u3** |
| deb | util-linux-extra | updated | 2.38.1-5+deb12u2 | **2.38.1-5+deb12u3** |
| python | virtualenv | updated | 20.27.1 | **20.29.1** |
| python | wheel | updated | 0.44.0 | **0.45.1** |
| python | xmltodict | updated | 0.13.0 | **0.14.2** |
| python | Pygments | added | not present | **2.19.1** |
| python | anyascii | added | not present | **0.3.2** |
| python | boltons | added | not present | **25.0.0** |
| python | fastjsonschema | added | not present | **2.21.1** |
| python | google-api-core | added | not present | **2.24.2** |
| python | google-auth | added | not present | **2.39.0** |
| python | google-cloud-secret-manager | added | not present | **2.23.3** |
| python | googleapis-common-protos | added | not present | **1.70.0** |
| python | grpc-google-iam-v1 | added | not present | **0.14.2** |
| python | grpcio | added | not present | **1.71.0** |
| python | grpcio-status | added | not present | **1.62.3** |
| python | markdown-it-py | added | not present | **3.0.0** |
| python | mdurl | added | not present | **0.1.2** |
| python | proto-plus | added | not present | **1.26.1** |
| python | protobuf | added | not present | **5.29.4** |
| python | pyasn1 | added | not present | **0.6.1** |
| python | pyasn1_modules | added | not present | **0.4.2** |
| python | rich | added | not present | **13.9.4** |
| python | rich-click | added | not present | **1.8.8** |
| python | rsa | added | not present | **4.9.1** |
| python | whenever | added | not present | **0.6.16** |
| python | exceptiongroup | removed | 1.2.2 | **not present** |
| python | pendulum | removed | 2.1.2 | **not present** |
| python | pytzdata | removed | 2020.1 | **not present** |
| python | tomli | removed | 2.1.0 | **not present** |

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=53033) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=53033) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=53033) it.</sub>

airbyteio committed:
Upgrade the base image to the latest version in metadata.yaml
PATCH bump source-typeform version to 1.4.2
Build source-typeform docker image for platform(s) linux/amd64, linux/arm64
Get dependency updates
Create or update pull request on Airbyte repository
Add changelog entry

dbgold17 commented:
/run-cat-tests

dbgold17 committed:
Merge branch 'master' into up-to-date/source-typeform

aaronsteers commented:
Acceptance test failure...

> <img width="627" alt="image" src="https://github.com/user-attachments/assets/3f18eb05-3d12-44c2-aa2f-b0ec960b15c8" />

[CAT Html report link](https://ff77d3812c84759bcf1c3cd8804df8bd63ee12048529bd2234efa01-apidata.googleusercontent.com/download/storage/v1/b/airbyte-ci-reports-multi/o/airbyte-ci%2Fconnectors%2Ftest%2Fpull_request%2Fup-to-date_source-typeform%2F1749077830%2F0a99c45298d53f6c42d02332e1882399d66fb84c%2Fsource-typeform%2F1.4.2%2Foutput.html?jk=AXbWWmnB3N_YRvslynpGTxwjPgeDyKYmpHCg3OMdL0Blr80T4TVnEPewJeIxY0wDVkTeDAOmHPvbo61uyet2wrmiYqrCwVMRMGsq6Yk9UZAaamjA0iFgoYPy5EkRoB1LzswjKkmOz3_knkyQq52HmFnZz0ocB3kkD5hVXKNY5SrINm2zuG3h3yA3fs5g28RfAyBU-cLYHlX6tI5mfvvq8O4-Eo5uvBYCJa8yv1bujHU2VTaVnXyTO2uVyoqBk0cTvsY4W9aB6OXecBJTwnEdfl20bPpx0iSZMxOzHbe8_Ngp2nJmnOXhAjLkCQ-EPyypJ_yF3xCIsOwUlWoXwi0InaOb5ktcBmnQ_70vykj_QeN3ya3alp8-N5x34XHNkXIBUc4KtxVgDlus0wjRMjpg-_H9avu7c2S1nSdIbsZPSIPBtZOuUCiWD2vScHTr3gvTmq1Tzl0ALbWOssmpTIUc5faqoM8vjT9OUB3D7fiAJ2BgQgFu5VHhlZ_VTJk7fFxufLbXwyY9AM7DOfZf9AmNT606rpcfqketWFGY9Imgs_5OISqDk3Ybrh8uBpWIyr9kWwUp4xPr6EMpOrEySFpXjUZansACqW5AlhTC09tHHnEdIfYm6fLdJCZWniiUY18q-u47fPXEH74MITJRNpCcAJpVqRoWvRIIcewP9HDpNTJ0m4WJ97b2wHQYg2qQ6P_0z8qIt7T3GClUUT40tO87OCrajXtHEydggK-JtmxiU6c_PTOR05fqOhKiHRTddHWuM6KRz9MiSm7558cMS96hEWoDhVkihXstXZ9UfkGvU1fccMd8lFvKckTY9O4RYdqGAD1ERfl1jTJHrntIGWsp5JvhysmMNlVBGA2kfPrQ90KUc8YMC3Wdte391j9AvBhyCyzU2SmcqPkSkRlhfHov62NcstW4nTnqN8MdRsuXQz0Lgac0BxC20lkl1lpFBIU85ZLrNWD_LEvtvPi-87l_yiqgVWYhvV_SjhWGwvQ_BPjGStToD7ldWRqeK7LP8Odurjg2mmKwMT452KBwwxxjZkwewlgHBB8m4Wqs9uD_ZLVWKfhb9Dh3b7NhZuou7sxT6OpVthCzkKZyuUrDfCwnMzy9xioYw1B1J0UAKK6nPjtIhjdviefFzrEAma_GMg2e9QlbyKpCZuel38_snNaQO6F29gRk_hcWOKdKfUVjpeb-AqiWL30RUu7CWPiBYLEmrmbPEuOTwVESsoppE--aXBjSRy8XpWY7nOwtgoX2GkXLvNNwwL2jkD_2JIqfwpzv3AY19HGQojbvrIfz4hkF2inHFDYtKqNsRAUCAEvhzXeYV5pq4a6eUxUsXHSm3KfF1aU0BJxQgWExi45sO45ZMgVDZ8wTXXBgxF7I90-GsxOv2GoRhE76Nkspvl8gfZtCvdYo46stbJGr8-wA22o4_5wxo8yBUFAIaeBNrm4_DpQ9jnQUIHez&isca=1)

Looks like test_check is failing.

> <img width="1109" alt="image" src="https://github.com/user-attachments/assets/5141c67a-0f9e-41b5-a2c5-a14fecceb54b" />

aaronsteers reviewed (approved):
Approving for force-merge, per @dbgold17's discretion.

The only failures appear to be related to stale auth or bad creds in the `oauth_config.json` config scenario.

If we can find/fix the issue with the auth, or at least confirm that this isn't a regression, then I think this is good to merge.

aaronsteers commented:
@dbgold17 - While I understand that `master` is already failing, that does get drift vs what is published as the image. You could locally try something like `docker run ... -v ... airbyte/source-typeform check --config=path/to/oauth_config.json` 

The challenge with that is just that you will need to get the syntax correct for volume-mapping the config from host to container volume.

dbgold17 commented:
ok I did a bit of digging here @aaronsteers 

Firstly, the command `docker run -v $(pwd)/secrets/:/data/ airbyte/source-typeform:latest check --config /data/config_oauth.json` fails with the same error that we see from integration tests and CATs
```
'Unable to connect to stream forms - 401 Client Error:  for url: https://api.typeform.com/oauth/token'
```
I checked out the credentials we use in that config and they appeared to be expired, so I went through the process of refreshing the access token, using the refresh token in the file, following the command described [here](https://www.typeform.com/developers/get-started/applications/#4-refresh-your-applications-tokens). However, that gave me an error that the refresh token was invalid. 

I went onto Airbyte Cloud and attempted to set up a typeform connection to see if I could get a new refresh token tied to our OAuth app. I noticed we only have one app and so it is probably shared between integration tests and our production app. Plugging that refresh token into the config and rerunning `docker run -v $(pwd)/secrets/:/data/ airbyte/source-typeform:latest check --config /data/config_oauth.json` gave me a successful check.

It seems that the process of refreshing tokens, also invalidates the old refresh token which is possible led to the issue in the first place. From their website:
> NOTE: For security purposes, the refresh procedure will invalidate the old refresh token. Therefore, your application won't be able to use the old refresh token anymore. Instead, your application should use the newly generated refresh token returned in the response to this query.


**next steps (let me know what you think)**
- I will store the new creds in the oauth_config on GSM, or wherever they live for fetching
- re-run the tests on the PR
- assuming they pass, we merge

dbgold17 committed:
Merge branch 'master' into up-to-date/source-typeform
